scripts/clip_interrogator_ext.py

```diff
@@ -202,25 +203,39 @@ def batch_process(folder, clip_model, mode, output_mode):
 
             # generate captions in first pass
             captions = []
+
             for file in files:
-                if shared.state.interrupted:
-                    break
-                image = Image.open(os.path.join(folder, file)).convert('RGB')
-                captions.append(ci.generate_caption(image))
-                shared.total_tqdm.update()
+                try:
+                    if shared.state.interrupted:
+                        break
+                    image = Image.open(os.path.join(folder, file)).convert('RGB')
+                    captions.append(ci.generate_caption(image))
+                except OSError:
+                    print(f" Could not read {file}; continuing")
+                    continue
+                finally:
+                    shared.total_tqdm.update()
 
             # interrogate in second pass
             writer = BatchWriter(folder, output_mode)
             shared.total_tqdm.clear()
             shared.total_tqdm.updateTotal(len(files))
             for idx, file in enumerate(files):
-                if shared.state.interrupted:
-                    break
-                image = Image.open(os.path.join(folder, file)).convert('RGB')
-                prompt = interrogate(image, mode, caption=captions[idx])
-                writer.add(file, prompt)
-                shared.total_tqdm.update()
-
+                try:
+                    if shared.state.interrupted:
+                        break
+                    image = Image.open(os.path.join(folder, file)).convert('RGB')
+                    try:
+                        prompt = interrogate(image, mode, caption=captions[idx])
+                        writer.add(file, prompt)
+                    except IndexError as e:
```

Comment:
Ah this is a problem. If we failed to load an image then the captions after that will be at the wrong locations. Instead of `IndexError` handling here it should add a blank string for the caption in the OSError exception handler above.

Reply:
Ok I did that and also moved the `captions.append` call into the `finally` block to make it a little DRYer